common/src/backends/otlp_tracing_bridge.rs

```diff
@@ -152,3 +153,104 @@ const fn severity_of_level(level: &Level) -> Severity {
         Level::ERROR => Severity::Error,
     }
 }
+
+pub struct ErrorTracingLayer<S> {
+    _registry: std::marker::PhantomData<S>,
+}
+
+impl<S> ErrorTracingLayer<S>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    pub fn new() -> Self {
+        ErrorTracingLayer {
+            _registry: std::marker::PhantomData,
+        }
+    }
+}
+
+impl<S> Layer<S> for ErrorTracingLayer<S>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+{
+    fn on_event(
+        &self,
+        event: &tracing_core::Event<'_>,
+        ctx: tracing_subscriber::layer::Context<'_, S>,
+    ) {
+        // We only care about error events.
+        if !ErrorVisitor::is_valid(event.metadata()) {
+            return;
+        }
+
+        let mut visitor = ErrorVisitor::default();
+        event.record(&mut visitor);
+
+        let DatadogError {
+            message,
+            r#type,
+            stack,
+        } = visitor.error;
+
+        if let Some(span) = ctx.lookup_current() {
+            if let Some(otel_data) = span.extensions_mut().get_mut::<OtelData>() {
```

Comment:
I can't find what sets the `OtelData` extension. Under which circumstance does it get set?

Reply:
It's set by the otel_layer in `setup_tracing`, so the error layer needs to go after it. I can leave a comment about that, as Jocelyn did in his valuable POC: https://github.com/shuttle-hq/shuttle/pull/1611/files#diff-6b6299a4e2fbb65c112d0cfaacb9227d9f15b1e1002d01c968b530d7c65d094c

Reply:
I believe this is the source: https://github.com/tokio-rs/tracing-opentelemetry/blob/v0.1.x/src/layer.rs#L911

Reply:
Ahhh, I see :+1:

Reply:
Yes, please comment this somewhere

Reply:
I added a comment in setup_tracing here: https://github.com/shuttle-hq/shuttle/pull/1626/commits/729eb6b1918b5ba8ee7ff13c488b63130e6c0672